engine/src/test/java/io/zeebe/engine/processor/workflow/multiinstance/MultiInstanceInputCollectionTest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.processor.workflow.multiinstance;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.zeebe.engine.util.EngineRule;
+import io.zeebe.model.bpmn.Bpmn;
+import io.zeebe.model.bpmn.BpmnModelInstance;
+import io.zeebe.protocol.record.intent.JobIntent;
+import io.zeebe.test.util.record.RecordingExporter;
+import io.zeebe.test.util.record.RecordingExporterTestWatcher;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class MultiInstanceInputCollectionTest {
+
+  @ClassRule public static final EngineRule ENGINE = EngineRule.singlePartition();
+
+  private static final String PROCESS_ID = "process";
+  private static final String JOB_TYPE = "test";
+  private static final String INPUT_COLLECTION = "items";
+  private static final String INPUT_ELEMENT = "item";
+
+  private static final BpmnModelInstance WORKFLOW =
+      Bpmn.createExecutableProcess(PROCESS_ID)
+          .startEvent()
+          .serviceTask(
+              "task",
+              t ->
+                  t.zeebeTaskType(JOB_TYPE)
+                      .multiInstance(
+                          b ->
+                              b.zeebeInputCollection(INPUT_COLLECTION)
+                                  .zeebeInputElement(INPUT_ELEMENT)))
+          .endEvent()
+          .done();
+
+  @Rule
+  public final RecordingExporterTestWatcher recordingExporterTestWatcher =
+      new RecordingExporterTestWatcher();
+
+  @Parameterized.Parameter(0)
+  public Collection<?> inputCollection;
+
+  @Parameterized.Parameters(name = "with input collection: {0}")
+  public static Object[][] parameters() {
+    return new Object[][] {
+      {Arrays.asList("a")},
+      {Arrays.asList(true, false)},
+      {Arrays.asList(10, 20, 30)},
+      {
+        Arrays.asList(
+            Collections.singletonMap("x", 1),
+            Collections.singletonMap("x", 2),
+            Collections.singletonMap("x", 3))
+      },
+      {Arrays.asList("x", null, true, 40)},
+    };
+  }
+
+  @Test
+  public void shouldCreateOneInstanceForEachElement() {
```

Comment:
Considering we're testing that we create one job/task per element in `MultiInstanceServiceTaskTest`, wouldn't it be better here to simply test that we create an element instance per element, and that it has the item set as a variable in its local scope?

Reply:
I'll change when implementing multi-instance sub-process.